pydarn/io/borealis/borealis_array.py

```diff
@@ -0,0 +1,482 @@
+        BorealisUtilities.check_arrays(self.filename, arrays, attribute_types,
+            dataset_types, unshared_fields)
+        self._arrays = arrays
+
+
+class BorealisArrayWrite():
+    """
+    Class for writing Borealis 'array' filetypes. Array files have
+    been restructured to no longer be read in record-by-record
+    style. 
+
+    See Also
+    --------
+    BorealisRawacf
+    BorealisBfiq
+    BorealisAntennasIq
+    BorealisRawrf
+
+    Attributes
+    ----------
+    filename: str
+        The filename of the Borealis HDF5 file being written.
+    borealis_filetype: str
+        The type of Borealis file. Restructured types include:
+        'bfiq'
+        'antennas_iq'
+        'rawacf'
+    compression: str
+        The type of compression to write the file as. Default zlib.
+    arrays: dict
+        The Borealis data in a dictionary of arrays, according to the 
+        restructured array file format.
+    """
+
+    def __init__(self, filename: str, borealis_arrays: dict, 
+                 borealis_filetype: str, compression: str = 'zlib'):
+        """
+        Write borealis records to a file.
+
+        Parameters
+        ----------
+        filename: str
+            Name of the file the user wants to write to
+        borealis_arrays: dict
+            borealis data dictionary. Keys are data field names and
+            unshared fields have a first dimension = number of records 
+            in the file.
+        borealis_filetype: str
+            The type of Borealis file. Restructured types include:
+            'bfiq'
+            'antennas_iq'
+            'rawacf'
+        compression: str
+            String representing compression type. Default zlib.
+        """
+        self.filename = filename
+        self.arrays = borealis_arrays
+        self.borealis_filetype = borealis_filetype
+        self.compression = compression
+        self.write_file()
+
+    def __repr__(self):
+        """For representation of the class object"""
+
+        return "{class_name}({filename}, {current_record_name})"\
+               "".format(class_name=self.__class__.__name__,
+                         filename=self.filename,
+                         current_record_name=self.current_record_name)
+
+    def __str__(self):
+        """For printing of the class object"""
+
+        return "Writing to filename: {filename} at record name: "\
+               "{current_record_name}".format(filename=self.filename,
+                    current_record_name=self.current_record_name)
+
+    def write_file(self) -> str:
+        """
+        Write Borealis records to a file given filetype.
+
+        Raises
+        ------
+        BorealisFileTypeError
+        """
+
+        if self.borealis_filetype == 'bfiq':
+            self.write_bfiq()
+        elif self.borealis_filetype == 'rawacf':
+            self.write_rawacf()
+        elif self.borealis_filetype == 'antennas_iq':
+            self.write_antennas_iq()
+        else:
```

Comment:
It is not currently written that way. What is the advantage of writing it that way instead of taking in the filetype at instantiation?

Reply:
Readability, searchability, flexibility, and maybe performance. 
1. string parameters that need to match are typo prone and the user may not know what to type without looking through documentation or if one has a good exception describing the issue. However, they would have to read something to get the answer instead of potentially discovering it (see 2).
2. Searchability, most text editors have autocompleted. So like myself and @aburrell I tend to use "." in classes to see what methods are available and then let the autocomplete type it out. This saves me time and potential typos. 
3. Code readability, it is easier to read in code `blag.write_fitacf()` and know what is happening in the code then seeing a class with several parameters and realize the one parameter `'fitacf'` was referencing to which filetype it was supposed to read in. 
4. Flexibility - if one wants to read in multiple files if they can use the methods vs. re-creating the class instance every loop call this may allow for better performance (may need to double-check this) and usability. The main case is also realtime feeds like reading map file data. Since the stream would be coming in every 2 minutes, re-creating the class instance which could hold some extra counter information might be a bit slow and annoying. 

Right now, I don't mind the text but making it only text input is inconsistent with superdarn.py.  
Again, we can see what the users prefer too.